python/kserve/kserve/storage/storage.py

```diff
@@ -272,8 +271,28 @@ def _download_s3(uri, temp_dir: str):
             if mimetype in ["application/x-tar", "application/zip"]:
                 Storage._unpack_archive_file(target, mimetype, temp_dir)
 
+    @staticmethod
+    def _download_hf(uri, temp_dir: str):
+        from transformers import AutoTokenizer, AutoConfig, AutoModel
```

Comment:
instead of loading model and then saving the model to local, which means you need resources (lot of memory) to load the model, maybe [snapshot_download](https://github.com/vllm-project/vllm/blob/65bf2ac165734fb6339210c4b2b8ce68d2391b77/vllm/model_executor/model_loader/weight_utils.py#L203-L210) SDK will be better.

Reply:
Hi @andyi2it , as mentioned above, [snapshot_download](https://huggingface.co/docs/huggingface_hub/v0.24.2/en/package_reference/file_download#huggingface_hub.snapshot_download) sdk will be the better option to download the model artifacts.